### ISSUE TITLE
Fail to build PyG Singularity image due to tzdata data-entry prompt

### ISSUE THREAD

**Issue by NucciTheBoss** [pyg-team/pytorch_geometric#4365]
### Top-level description

Container build with singularity using the recipe provided in `pytorch_geometric/docker` fails to build because `tzdata` makes a data-enter request during the build process. The process does not resume after entering the selected timezone.

### Copy of stacktrace
```
Setting up tzdata (2022a-0ubuntu0.20.04) ...
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76.)
debconf: falling back to frontend: Readline
Configuring tzdata
------------------

Please select the geographic area in which you live. Subsequent configuration
questions will narrow this down by presenting a list of cities, representing
the time zones in which they are located.

  1. Africa      4. Australia  7. Atlantic  10. Pacific  13. Etc
  2. America     5. Arctic     8. Europe    11. SystemV
  3. Antarctica  6. Asia       9. Indian    12. US
Geographic area:
```

### Potential fix

Luckily, this is a frequent issue encountered during the container build process (especially when using Ubuntu as base image). What needs to be done is that `/etc/timezone` will need to be created and set during build time.

### Potential issues related to this one

If this is an issue for the singularity image, then this will most likely be an issue for the Docker image as well.

### Environment

* Singularity version: 3.8.5
* Any other relevant information (*e.g.*, version of `torch-scatter`):
  * Version of OS running singularity: 
```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Comment by rusty1s** [comment]
Thanks for reporting. Our `Dockerfile` is pretty old and desperately needs an update soon :)

Do you have interest in contributing your fix?

FYI @aeli1

**Comment by NucciTheBoss** [comment]
@rusty1s Yes - I have a fix in the works :). I will contribute it after I have some folks on my team test the new recipe.

My plan is to fix the current image just so that it works, push the fix, and then open another pull request updating the `Dockerfile` and `Singularity` recipe to the current version of pytorch geometric.

**Comment by NucciTheBoss** [comment]
Pull request #4376 resolves the issue with the tzdata data-entry prompt. Going to close out this issue since original problem is solved. Hoping to follow up with newer `Dockerfile` and `singularity` recipe in the next couple of days.
